test/promise.js

```diff
@@ -0,0 +1,88 @@
+
+var Router = require('..')
+var utils = require('./support/utils')
+var Promise = require('bluebird')
+
+var assert = utils.assert
+var createServer = utils.createServer
+var request = utils.request
+
+describe('Promise', function () {
+  it('rejecting will trigger error handlers', function (done) {
+    var router = Router()
+    var server = createServer(router)
+
+    router.use(function (req, res) {
+      return new Promise(function (resolve, reject) {
+        reject(new Error('Happy error'))
+      })
+    })
+
+    request(server)
+    .get('/')
+    .expect(500, done)
+  })
+
+  it('will be ignored if next is called', function (done) {
+    var router = Router()
+    var server = createServer(router)
+    var count = 0
+
+    router.use(function (req, res, next) {
+      count++
```

Comment:
This method is especially useful at places like L36 where I can test a closure has *not* been called. Is there a way to do this with the `x-hit` stuff?